### ISSUE TITLE
Problems installing Liota on a Python virtualenv

### ISSUE THREAD

**Issue by richia44** [vmware/liota#85]
Hello

We started working with Liota a few weeks ago, developing my own packages and sending information to my own DCC. So far, We managed to send data through HTTP but since you added the MQTT implementation we tried to work with it. Our problem comes with the Liota installation through pip, we got some errors because our packages with MQTT implementation don't find the MQTT files from Liota (can't resolve the imports), and our package files are on the package folder inside the installed Liota, are we doing maybe something wrong?

We are used to develop using virtualenv and the idea is continue the same process with Liota unless you think it's a bad idea. We've created our own requirements.txt file to create de virtualenv based on the file included with Liota, but I decided to download Liota from Github in order to have the latest Liota, work with MQTT and resolve the import issue. But through this method, Liota doesn't install correctly, doesn't appear on the virtualenv libraries and I get "ImportError: No module named liota.core.package_manager" each time I try to run the liotad.py file. Any advice on how can install Liota on a virtualenv and use MQTT? 

P.D: this is how mi requirements.txt looks like
aenum==1.4.5
appdirs==1.4.0
linux-metrics==0.1.4
packaging==16.8
paho-mqtt==1.2
Pint==0.7.2
pyparsing==2.1.10
six==1.10.0
websocket-client==0.37.0
-e git+https://github.com/vmware/liota.git@e2427c533db363aeccb5ad34c5b37e97330db06f#egg=liota

**Comment by Venkat2811** [comment]
Hello @richia44 ,

Thank you for using LIOTA !

1. "Our problem comes with the Liota installation through pip, we got some errors because our packages with MQTT implementation don't find the MQTT files from Liota (can't resolve the imports)".

`Ans:` LIOTA installation through `pip` is not updated with MQTT support yet.  We are working on few changes and providing more examples using MQTT #71 .  It will be updated soon!

2. ImportError in virtualenv

`Ans:`  Would suggest you to refer [this](http://stackoverflow.com/questions/15048316/pip-install-into-virtualenv-from-github-using-e-option-fails-to-add-package-to) and try resolving the issue.  If you still face the issue let us know and we'll help you.

Thanks,
Venkat

**Comment by KohliDev** [comment]
@richia44 We`ll be coming up with new liota pip version which will include MQTT by this week.

**Comment by richia44** [comment]
@KohliDev Thank you very much for your work. By the way, are there any plans to implemente MQTT for receiving info from the QMTT broker server? seems like Liota is designed to send data to the DCCs, but it would be great if we can also receive operations to have some control on the devices (like reboot, shutdown  or similar). Thanks again.

**Comment by KohliDev** [comment]
Yes, we have it in plans as part of actions. We have also included liota-resp/generated_local_uuid_of_edge_system
topic in our auto-generated MQTT topic list which can be used to receive the response over MQTT for devices & actuators. The auto-subscription to the liota-resp/* topic will be included in the MQTT code soon.
